File: 코딩테스트 대비/1197.cpp
#include <iostream>
#include <queue>

using namespace std;

int main()
{
   int n, m, a, b, c;
   cin >> n >> m;
   vector<int> check(n);
   for (int i = 0; i < n; ++i)
      check[i] = i;
   priority_queue<vector<int>> q;
   for (int i = 0; i < m; ++i)
   {
      cin >> a >> b >> c;
      q.push({-c, a - 1, b - 1});
   }

   int answer = 0;
   for (int i = 0; i < n - 1;)
   {
      if (check[q.top()[1]] != check[q.top()[2]])
      {
         int t = check[q.top()[2]];
         for (int i = 0; i < n; ++i)
            if (check[i] == t)
               check[i] = check[q.top()[1]];
         answer -= q.top()[0];
         ++i;
      }
      q.pop();
   }
   cout << answer;

   return 0;
}

File: 코딩테스트 대비/1197.py
from sys import stdin
from queue import PriorityQueue

n, m = map(int, stdin.readline().split())
q = PriorityQueue()
for _ in range(m):
    a, b, c = map(int, stdin.readline().split())
    q.put([c, a, b])

v = list(range(n + 1))
def getParent(n):
    if v[n] == n:
        return n
    v[n] = getParent(v[n])
    return v[n]

answer = 0
while q.empty() == False:
    t = q.get()
    if getParent(t[1]) != getParent(t[2]):
        answer += t[0]
        v[getParent(t[2])] = getParent(t[1])
print(answer)